File: DSA/Big O/Python/04-O(n).py
list_1 = [1,2,3,4]
list_2 = [1,2,3,4,5]
list_3 = [1,2,3,4,5,6]

def my_function(l_list):
    for i in l_list:
        print(i)

my_function(list_1) # O(4)
my_function(list_2) # O(5)
my_function(list_3) # O(6)
# final ans is: it is O(n) function

# O(n) - Linear Time - Fair
# in O(n), input element == operation
# we can write O(n) or O(x) or O(fish) or O(anything)
# O(n) here O means 'Big O' & n means 'number of input'
# O(n) means n times operation

File: DSA/Big O/JS/04-O(n).js
const lapArray = new Array(100).fill("dell")

function findLaptop(array){
    for (i=0; i<array.length; i++){
        if (array[i] == "dell"){
            console.log("found dell")
        }
    }
}

findLaptop(lapArray) // O(n) - Linear time

// O(n) because Operation is depends on Input
// O(n) means n times operation
